File: 10_Sorting_MaxItemsMinValue.py
"""
https://www.hackerrank.com/challenges/mark-and-toys
"""

# Complete the maximumToys function below.
def maximumToys(prices, k):
    prices.sort()
    i = 0
    mysum = 0
    for e in prices:
        mysum += e
        if mysum <= k:
            i += 1
    print(i)




prices = [1, 12, 5, 111, 200, 1000, 10]
k = 50
maximumToys(prices, k)

File: 10_Sorting_MaxItemsMinValue.cpp
//
// Created by jan on 29.12.19.
// https://www.hackerrank.com/challenges/mark-and-toys
//

#include <bits/stdc++.h>

using namespace std;

vector<string> split_string(string);

// Complete the maximumToys function below.
int maximumToys(vector<int> prices, int k) {
    sort (prices.begin(), prices.end());
    int i = 0;
    long mysum = 0;
    for (int j = 0; j < prices.size(); j++){
        mysum += prices[j];
        if (mysum < k){
            i++;
        }
    }
    return(i);
}